server/src/main/java/org/elasticsearch/common/bytes/CompositeBytesReference.java

```diff
@@ -225,4 +225,15 @@ public void writeTo(OutputStream os) throws IOException {
     public long ramBytesUsed() {
         return ramBytesUsed;
     }
+
+    @Override
+    public int getIntLE(int index) {
+        int i = getOffsetIndex(index);
+        int idx = index - offsets[i];
+        int end = idx + 4;
+        if (end <= references[i].length()) {
```

Comment:
NIT: Maybe store `references[i]` to a variable? That should be a little faster here on average I think since I'd expect the fast path to be taken most of the time.

Reply:
This one helps!

Without:
```
Benchmark                      (type)  Mode  Cnt  Score   Error  Units
IntArrayBenchmark.read  composite_1mb  avgt    7  8.131 ± 0.138  ns/op
```

With:
```
Benchmark                      (type)  Mode  Cnt  Score   Error  Units
IntArrayBenchmark.read  composite_1mb  avgt    7  7.770 ± 0.140  ns/op
```